Step02_Feature1_FixTheCompilationError/testcases._cpp

```diff
@@ -0,0 +1,24 @@
+#define CATCH_CONFIG_MAIN 
+
+#include "test/catch.hpp"
+#include "currentMeasurements.h"
+
+/* FEATURE 1 : The return Value for {4,5} from getRangeDetails should be 1 .
+   TDD_STEP1 : Write the Test Code(Unit Test) without having the implementation of a function being called .
+   Actions : Write only test code (Unit Test).
+   Result : FAILED .
+   Reason : There is no function defined present     */
+
+/* FEATURE 1 : The return Value for {4,5} from getRangeDetails should be 1 .
+   TDD_STEP2 : write enough code so that unit test code can compile and execute, but will not PASS .
+   Actions : Define a function but not implementation , And Tell Somehow the function is not implemented by .
+         1.  Forward declaration .
+         2.  Throws Exception .
+         3.  If language has some constraints return a value which tells function is not implemented .
+   Result : FAILED ... We conclude Bug Found.
+   Reason : Function is defined , but there is no implementation for the called function and function is made to return 0 always  */
+   
+TEST_CASE("Get the range distribution details for consecutive numbers scenario 1"){
+    int sampleNumbersArray[] = {4,5};

```

Comment:
As per the test case, the expectation of production code is that it has to handle only a single range.
The test cases need to be added to verify expectation when various type of inputs are provided... for ex. 1,2,3,5,7, 9, 12

More Well defined test cases we have, better control on the preservation of behavior of production code and its extension. Maybe you can still define test cases with different inputs.

Reply:
Hello Sir,
This folders Step01 ... Step02 etc is used to demonstrate the different steps involved in TDD .
So, these test scripts are not complete ..
The final result can be seen in the scripts present in root folder of the project test-tdd.cpp

Reply:
👍